proj-xcode/Classes/keychain/PA2Keychain.m

```diff
@@ -264,33 +264,112 @@ - (NSDictionary*) allItemsWithPrompt:(NSString*)prompt withStatus: (OSStatus *)s
 
 #pragma mark - Biometry support
 
-+ (BOOL) canUseBiometricAuthentication
-{
-	return [self supportedBiometricAuthentication] != PA2SupportedBiometricAuthentication_None;
-}
+/**
+ Looks like Apple introduced a new biometry type. We should try to continue,
+ and pretend that TouchID is available. Application's UI will probably display
+ wrong information, but at least it may work.
+ */
+#define __BiometricAuthenticationType_Fallback 	PA2BiometricAuthenticationType_TouchID
 
-+ (PA2SupportedBiometricAuthentication) supportedBiometricAuthentication
+/**
+ A private function returns full information about biometric support on the system. The method internally
+ uses `LAContext.canEvaluatePolicy()`.
+ */
+static PA2BiometricAuthenticationInfo _getBiometryInfo()
 {
+	PA2BiometricAuthenticationInfo info = { PA2BiometricAuthenticationStatus_NotSupported, PA2BiometricAuthenticationType_None };
 #if !defined(PA2_EXTENSION_SDK)
 	// Check is available only for regular iOS applications
 	if (@available(iOS 9, *)) {
 		LAContext * context = [[LAContext alloc] init];
-		BOOL canEvaluate = [context canEvaluatePolicy:kLAPolicyDeviceOwnerAuthenticationWithBiometrics error:nil];
+		NSError * error = nil;
+		BOOL canEvaluate = [context canEvaluatePolicy:kLAPolicyDeviceOwnerAuthenticationWithBiometrics error:&error];
 		if (canEvaluate) {
+			// If we can evaluate, then everything is quite simple.
+			info.currentStatus = PA2BiometricAuthenticationStatus_Available;
+			// Now check the type of biometry
 			if (@available(iOS 11.0, *)) {
 				LABiometryType bt = context.biometryType;
 				if (bt == LABiometryTypeTouchID) {
-					return PA2SupportedBiometricAuthentication_TouchID;
+					info.biometryType = PA2BiometricAuthenticationType_TouchID;
 				} else if (bt == LABiometryTypeFaceID) {
-					return PA2SupportedBiometricAuthentication_FaceID;
+					info.biometryType = PA2BiometricAuthenticationType_FaceID;
+				} else {
```

Comment:
We should probably have one more `else if` branch here, for `LABiometryTypeNone` value (see the [docs](https://developer.apple.com/documentation/localauthentication/labiometrytype?language=objc)). Currently, we may fallback to unknown biometry type even for this value (if, for some reason, `canEvaluate` is true at the same time)...

Reply:
That case will never happen, because biometry is already evaluated and "none" type makes no sense here.